### ISSUE TITLE
InceptionResNetV2 : is the unrolling wrong ?

### ISSUE THREAD

**Issue by AndreaLanfranchi** [serengil/deepface#1143]
I believe the `InceptionResNetV2` function has been unrolled wrongly.
Consider [this code](https://github.com/serengil/deepface/blob/329606ffe83d5dd2b0ed97bcb1950cd010c93ea2/deepface/basemodels/Facenet.py#L1284-L1289) :
```python
   [...]
    # 5x Block8 (Inception-ResNet-C block):

    branch_0 = Conv2D(
        192, 1, strides=1, padding="same", use_bias=False, name="Block8_1_Branch_0_Conv2d_1x1"
    )(x)
    branch_0 = BatchNormalization(
        axis=3,
    [...]
```

As the comment says (and as I've found in other implementations) the block-c should be repeated 5 times.
However if we follow the repetitions we can see it goes **through 6 iterations** as [shown here](https://github.com/serengil/deepface/blob/329606ffe83d5dd2b0ed97bcb1950cd010c93ea2/deepface/basemodels/Facenet.py#L1656-L1663)
```python
   [...]
    mixed = Concatenate(axis=3, name="Block8_6_Concatenate")(branches)
    up = Conv2D(1792, 1, strides=1, padding="same", use_bias=True, name="Block8_6_Conv2d_1x1")(
        mixed
    )
    up = Lambda(scaling, output_shape=K.int_shape(up)[1:], arguments={"scale": 1})(up)
    x = add([x, up])

    # Classification block
    [...]
```

My strong recommendation is to never unroll such complex loops.

**Comment by serengil** [comment]
In that case, we should not be able to load the model weights.

**Comment by AndreaLanfranchi** [comment]
Not sure I follow.

**Comment by serengil** [comment]
Maybe the confusion is because of that FaceNet models has the backbone of InceptionResNetV1 instead of V2: https://github.com/serengil/deepface/blob/master/deepface/basemodels/Facenet.py#L1673

**Comment by AndreaLanfranchi** [comment]
That's a label ... not a value.
And how is the function returning the model is named "V2" ?

**Comment by serengil** [comment]
should be v1

**Comment by AndreaLanfranchi** [comment]
Are you sure ? It seems incredibly matching V2 impl.
Used [this](https://github.com/skaldek/InceptionResNetV2/blob/master/Inception.py) and [this](https://github.com/skaldek/InceptionResNetV2/blob/master/modules.py) as comparison.

**Comment by AndreaLanfranchi** [comment]
You also wrote me earlier that you unrolled [this impl](https://github.com/yuyang-huang/keras-inception-resnet-v2/blob/master/inception_resnet_v2.py) (which is V2)

**Comment by serengil** [comment]
Long story short, I do not expect anything wrong in the model. If that would be possible (additional or missing layers), then pre-trained weight cannot be loaded.

Besides, FaceNet-512d and FaceNet-128d models are the most robust ones in deepface's portfolio according to the accuracy scores.

**Comment by serengil** [comment]
Do you mind to close this?

**Comment by AndreaLanfranchi** [comment]
Closing as addressed in #1144
